build.py

```diff
@@ -123,13 +120,18 @@ def __updateDigest( project, config ) :
 		with open( patch ) as f :
 			config["digest"].update( f.read() )
 
-def __loadConfigs( variables ) :
+def __loadConfigs( variables, variants ) :
 
-	# Load configs
+	# Load configs and apply variants and platform overrides.
 
 	configs = {}
 	for project in __projects() :
-		configs[project] = __loadConfig( project )
+		configs[project] = __loadJSON( project )
+		if project in variants :
+			__applyConfigOverrides( configs[project], "variant:{}".format( variants[project] ) )
+		for variantProject, variant in variants.items() :
+			__applyConfigOverrides( configs[project], "variant:{}:{}".format( variantProject, variant ) )
+		__applyConfigOverrides( configs[project], "platform:osx" if sys.platform == "darwin" else "platform:linux" )
```

Comment:
🦴 🐕  Feels like we could do with some documentation on how variants work at this point, ie the two version of `variant:` keys that can appear in a project's config and how to then invoke `build.py`.  I know we generally go for 'see the src as an example', but it does precipitate grepping for working out which projects do use it. It would also make it easier to review if the logic here is doing what is expected.

Reply:
c1fcbf0

Reply:
Epic, thanks. Thats great.